### PR TITLE
Appnexus: Update appendMemberId function

### PR DESCRIPTION
**Pull request by Sonali-More-Xandr**
Update appendMemberId function to use net/url to append query parameter member_id. Currently, we are checking if there is already a query parameter in the uri string by using string.contains() function. This PR updates this function to use net/url package to append the member_id to the url.

### CODE REVIEW

File: adapters/appnexus/appnexus.go [hunk 1]
```diff
@@ -403,11 +404,11 @@ func (a *adapter) findIabCategoryForBid(bid *appnexusBidExt) (string, bool) {
 }
 
 func appendMemberId(uri string, memberId string) string {
-	if strings.Contains(uri, "?") {
-		return uri + "&member_id=" + memberId
-	}
-
-	return uri + "?member_id=" + memberId
+	u, _ := url.Parse(uri)
```
Comment (SyntaxNode) [inline adapters/appnexus/appnexus.go:407]: Please do not ignore errors. I recommend moving `url.Parse` into the `Builder` method. You can safely make a shallow clone of the `URL` object here instead and proceed with the query string manipulation.

**Review by gargcreation1992 — COMMENTED** [review]
g2g after Scott's comment!

File: adapters/appnexus/appnexus.go (hunk 1 above)
Comment (Sonali-More-Xandr) [reply adapters/appnexus/appnexus.go]: Thank you for the suggestion, Scott! I have made the changes in [61090f7](https://github.com/prebid/prebid-server/pull/2747/commits/61090f76f85cced2fff19791b0e43c06b45e2fba)

**Review by gargcreation1992 — DISMISSED** [review]

File: adapters/appnexus/appnexus_test.go [hunk 2]
```diff
@@ -24,10 +25,14 @@ func TestJsonSamples(t *testing.T) {
 }
 
 func TestMemberQueryParam(t *testing.T) {
-	uriWithMember := appendMemberId("http://ib.adnxs.com/openrtb2?query_param=true", "102")
-	expected := "http://ib.adnxs.com/openrtb2?query_param=true&member_id=102"
-	if uriWithMember != expected {
-		t.Errorf("appendMemberId() failed on URI with query string. Expected %s, got %s", expected, uriWithMember)
+	uri, err := url.Parse("http://ib.adnxs.com/openrtb2?query_param=true")
```
Comment (mansinahar) [inline adapters/appnexus/appnexus_test.go:28]: **Nitpick:** Consider modifying this function to use `assert` for assertions and rename the function to `TestAppendMemberID`. As a rule of thumb I suggest we name test functions as actual function name it's testing with the `Test` prefix. I understand your PR isn't the one that added this test function so this is not to blame you :)

```
func TestAppendMemberID(t *testing.T) {
	uri, err := url.Parse("http://ib.adnxs.com/openrtb2?query_param=true")
	assert.NoError(t, err, "Failed to parse URI with query string")
	uriWithMember := appendMemberId(*uri, "102")
	expected := "http://ib.adnxs.com/openrtb2?member_id=102&query_param=true"
	assert.Equal(t, expected, uriWithMember.String(), "Failed to append member id to URI with query string")
}
```

Comment (Sonali-More-Xandr) [reply adapters/appnexus/appnexus_test.go:28]: updated the name and assertions in [040e2fb](https://github.com/prebid/prebid-server/pull/2747/commits/040e2fbd75507b7ffaa79d8a17f23f5e2148ecfc)

**Review by mansinahar — DISMISSED** [review]

**Review by mansinahar — APPROVED** [review]

**Review by gargcreation1992 — APPROVED** [review]

**Pull request merged**